.circleci/config.yml

```diff
@@ -0,0 +1,96 @@
+version: 2.1
+
+workflows: 
+  code-commit:
+    jobs:
+      - black-python-code-formatter:
+          context: org-sightmachine
+      - run-unit-tests:
+          context: org-sightmachine
+
+orbs:
+  gcp-gke: circleci/gcp-gke@0.2.0
+  python: circleci/python@2.1.1
+
+commands:
+  disable_docker_hub:
+    description: "Disable Docker Hub access for running container/VM."
+    steps:
+      - run:
+          name: Disable Docker Hub
+          command: |
+            mkdir -p ~/.docker/
+            echo '{"auths":{"https://index.docker.io/v1/":{"auth":"YTpiYWQK"}}}' > ~/.docker/config.json
+  disable_build_if_not_pr:
+    description: "Stop the current build if not in a PR context."
+    steps:
+      - run:
+          name: Abort if not building a pull request
+          command: |
+            if [ -z "$CIRCLE_PULL_REQUEST" ]; then
+              circleci-agent step halt
+            fi
+
+
+jobs:
+  ## ------------------ Run Unit Tests ------------------
+
+  run-unit-tests:
+    # Use VM over docker container
+    machine:
+      # # https://circleci.com/docs/2.0/configuration-reference/#available-machine-images
+      # image: ubuntu-2204:2022.07.1
+    docker:
+      - image: cimg/python:3.11.0
+    steps:
+      # Make sure that the build doesn't accidentally pull images from Docker Hub.
+      # Pulls from Docker Hub may fail due to rate-limiting after Nov 1 2020.
+      - disable_docker_hub
+      - attach_workspace:
+          at: /tmp/workspace
+
+      # - configure_environment
+      - checkout
+      - python/install-packages:
+          pkg-manager: pip
+      - run:
+          name: Run Unit Tests
+          command: |
+            mkdir test-results
+            coverage run -m pytest --junitxml=test-results/junit.xml tests
```

Comment:
May want to add a `.pytest.ini` file to configure xunit1 format.
https://docs.pytest.org/en/7.1.x/reference/customize.html#pytest-ini
https://github.com/sightmachine/ma/blob/master/pytest.ini